test/util/createVersionId.test.ts

```
@@ -0,0 +1,173 @@
+import {describe, expect, it} from 'vitest'
+
+import {type SanityDocumentStub} from '../../src/types'
+import {
+  deriveDocumentVersionId,
+  generateReleaseId,
+  getDocumentVersionId,
+} from '../../src/util/createVersionId'
+
+describe('createVersionId', () => {
+  describe('generateReleaseId', () => {
+    it('generates a string with 8 characters', () => {
+      const releaseId = generateReleaseId()
+      expect(typeof releaseId).toBe('string')
+      expect(releaseId.length).toBe(8)
+    })
+
+    it('generates unique values on multiple calls', () => {
+      const ids = new Set<string>()
+      for (let i = 0; i < 100; i++) {
+        ids.add(generateReleaseId())
+      }
+      expect(ids.size).toBe(100)
+    })
+
+    it('only uses characters from the defined alphabet', () => {
+      const releaseId = generateReleaseId()
+      const validChars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+      releaseId.split('').forEach((char) => {
+        expect(validChars.includes(char)).toBe(true)
+      })
+    })
+  })
+
+  describe('getDocumentVersionId', () => {
+    it('returns a draft ID when no releaseId is provided', () => {
+      const publishedId = 'document123'
+      const versionId = getDocumentVersionId(publishedId)
+      expect(versionId).toBe('drafts.document123')
+    })
```

This feels like it could confuse some users, what if instead of returning a draft id we throw?

---

nvm. it's ok, I see that you use it as part of the createVersion function which accepts either a version id or nothing, which will create a draft. So you need this to be undefined